custom_components/hacs/base.py

```diff
@@ -876,6 +878,11 @@ async def async_get_category_repositories_experimental(self, category: str) -> N
                 continue
             if repo in self.common.archived_repositories:
                 continue
+            try:
+                V2_REPO_SCHEMA[category](repo_data)
+            except vol.Invalid as exception:
+                self.log.info("Got invalid data for %s (%s)", repo, exception)
+                continue
```

Comment:
As discussed on Discord, it would be more natural if the validation and discarding of invalid data happened in the data client. However, the data client is used also by `scripts/data/generate_category_data.py` and it may be confusing if that script doesn't get the full data.